pixano/app/api/items.py

```diff
@@ -338,28 +338,40 @@ async def get_dataset_item(  # noqa: D417
     NoneMask = CompressedRLE.none()
     NoneKeypoints = KeyPoints.none()
 
-    # Note: This is used to find the entity of an annotation
-    # (see "Entity features" part below)
-    # this could be replaced by new python API get_data(...?)
-    # But, it is used for retrieving features at various levels
-    # and gather them at annotation level...
-    # May need to rework / think more about this
-    def findEntity(ann: Annotation) -> Entity | None:
-        for group_entity in groups[_SchemaGroup.ENTITY]:
-            entity = next(
-                (entity for entity in getattr(item, group_entity) if entity.id == ann.entity_ref.id),
-                None,
-            )
-            if entity:
-                return entity
-        return None
+    def findTopEntity(ann: Annotation) -> Entity | None:
+        """Find top entity of an annotation.
+
+        Args:
+            ann (Annotation): annotation
+
+        Returns:
+            Entity | None: top Entity for this annotation
+        """
+        try:
+            return findTopParentEntity(ann.entity)
+        except ValueError:
+            return None
+
+    def findTopParentEntity(entity: Entity) -> Entity | None:
```

Comment:
This is not how we should name functions in python => find_top_parent_entity however i've seen that other part of the API have this kind of naming functions.

Should we adress them now or when we will refactor the front and API  ?

Reply:
you right, corrected